src/hooks/useMinimumAmountGuard.ts

```diff
@@ -0,0 +1,43 @@
+import type { RootState } from "../store";
+import { useMemo } from "react";
+import { useSelector } from "react-redux";
+import { ChainId, isEVMChain } from "@certusone/wormhole-sdk";
+
+function checkIfIsBelowMinimum(amount: string, decimals: number) {
+  try {
+    const divider = Math.pow(10, decimals);
+    const floatAmount = parseFloat(amount);
+    const intAmount = floatAmount * divider;
+    return Math.trunc(intAmount) <= 0;
+  } catch (err: any) {
+    console.error(err);
+    return true;
+  }
+}
+
+function getAdjustedDecimals(
+  chainId: ChainId,
+  isNativeAsset: boolean,
+  decimals: number
+) {
+  return isEVMChain(chainId) && !isNativeAsset && decimals > 8
+    ? decimals - 8
+    : decimals;
+}
+
+export default function useMinimumAmountGuard() {
+  const {
+    amount,
+    sourceChain,
+    sourceParsedTokenAccount: { decimals = 0, isNativeAsset = false } = {},
+  } = useSelector((state: RootState) => state.transfer);
```

Comment:
@sebastianscatularo can we pass to the custom hook the variables needed from outside? so we do not have to use `useSelector` from the inside? I think will be more clear.

What do you think?

Reply:
yep, I agree, I'll update it